src/app/src/components/annotations/menu.jsx

```diff
@@ -360,29 +364,40 @@ export default class AnnotationMenu extends Component {
             onClick={this.props.callbacks.OpenFileManagement}
           />
           <MenuDivider title="Inference" />
+
           {this.props.predictDone === 0 ? (
-            <>
-              <MenuItem
-                icon="redo"
-                text="Re-Analyse"
-                label={<KeyCombo combo="R" />}
-                className={
-                  this.props.userEditState === "Re-Analyse" ? "bp3-active" : ""
-                }
-                onClick={() => this.props.callbacks.GetInference()}
-              />
-              <MenuItem
-                icon="heat-grid"
-                text="Bulk Analysis"
-                label={<KeyCombo combo="B" />}
-                className={
-                  this.props.userEditState === "Bulk Analysis"
-                    ? "bp3-active"
-                    : ""
-                }
-                onClick={() => this.props.callbacks.GetInference()}
-              />
-            </>
+            <Tooltip
+              content="Load model before analysing"
+              position={Position.TOP}
+              disabled={this.props.isConnected && this.props.loadedModel}
+            >
+              <div className={classes.InferenceMenuItem}>
+                <MenuItem
+                  disabled={!this.props.isConnected || !this.props.loadedModel}
+                  icon="redo"
+                  text="Re-Analyse"
+                  label={<KeyCombo combo="R" />}
+                  className={
+                    this.props.userEditState === "Re-Analyse"
+                      ? "bp3-active"
+                      : ""
+                  }
+                  onClick={() => this.props.callbacks.GetInference()}
+                />
+                <MenuItem
+                  disabled={!this.props.isConnected || !this.props.loadedModel}
+                  icon="heat-grid"
+                  text="Bulk Analysis"
+                  label={<KeyCombo combo="B" />}
+                  className={
+                    this.props.userEditState === "Bulk Analysis"
+                      ? "bp3-active"
+                      : ""
+                  }
+                  onClick={() => this.props.callbacks.GetInference()}
```

Comment:
Why is bulk analysis and single inference pointing towards the same function? 

Note: ignore this if this is addressed in another PR.

Reply:
Yeap, it's in another PR that's not merged yet

Reply:
Got it!